wcfsetup/install/files/lib/system/search/SearchResultHandler.class.php

```diff
@@ -138,6 +138,23 @@ public function getTemplateName(): array
         return $this->getLegacyTemplateName();
     }
 
+    /**
+     * @return string[]
+     */
+    public function getCustomIcons(): array
+    {
+        $customIcons = [];
+        foreach (SearchEngine::getInstance()->getAvailableObjectTypes() as $name => $type) {
+            if ($type instanceof ISearchProvider) {
+                if ($type->getCustomIconName()) {
+                    $customIcons[$name] = $type->getCustomIconName();
+                }
```

Comment:
```suggestion
                $customIcons[$name] = $type->getCustomIconName();
```
If null is used, the check isn't necessary, because the suggested `isset()` in the template will implicitly check for null.